CMakeLists.txt

```diff
@@ -29,16 +29,30 @@ find_package(catkin REQUIRED COMPONENTS roscpp)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
-add_library(Log++ ${CMAKE_CURRENT_SOURCE_DIR}/include/log++.h)
+if(catkin_FOUND)
+        catkin_package(
+                INCLUDE_DIRS include
+                LIBRARIES lpp
+                CATKIN_DEPENDS roscpp
+        )
+endif()
+add_library(lpp ${CMAKE_CURRENT_SOURCE_DIR}/include/log++.h)
 # As there are no src files, set the linker language explicitly
-set_target_properties(Log++ PROPERTIES LINKER_LANGUAGE CXX)
+set_target_properties(lpp PROPERTIES LINKER_LANGUAGE CXX)
 
 add_executable(lpp_demo ${CMAKE_CURRENT_SOURCE_DIR}/src/demo.cpp)
 target_compile_definitions(lpp_demo PRIVATE MODE_LPP)
 
 if (catkin_FOUND)
+        set(catkin_INCLUDE_DIRS ${catkin_INCLUDE_DIRS} ${LPP_INCLUDE_DIRS})
         include_directories(${catkin_INCLUDE_DIRS})
         target_link_libraries(lpp_demo ${catkin_LIBRARIES})
+
+        install(TARGETS lpp
+                ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+                LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+                RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
+                )
```

Comment:
does this work if we don't use install spaces?

Reply:
Yes because all env variables are set by default according to the catkin cmake documentation